packages/transaction-messages/src/decompile-message.ts

```diff
@@ -241,9 +241,10 @@ export function decompileTransactionMessage(
         createTransactionMessage({ version: compiledTransactionMessage.version as TransactionVersion }),
         m => setTransactionMessageFeePayer(feePayer, m),
         m =>
-            instructions.reduce((acc, instruction) => {
-                return appendTransactionMessageInstruction(instruction, acc);
-            }, m),
+            instructions.reduce(
+                (acc, instruction) => appendTransactionMessageInstruction(instruction, acc),
+                m as BaseTransactionMessage & TransactionMessageWithFeePayer,
```

Comment:
Hunh. Was this not already the case? Why did you have to cast it to `BaseTransactionMessage & TransactionMessageWithFeePayer`?

Also, is there a need to `TransactionMessageWithFeePayer<typeof feePayer>` or no?

Reply:
Ahhh. Got it.